packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -573,38 +628,30 @@ export class TempCollabSpaceRuntime
 		return this.createCollabChannel(value, channelId);
 	}
 
-	private mapChannelToCell(channelId: string) {
+	private parseChannelId(channelId: string): { rowId: string; colId: string; iteration: string } {
 		const parts = channelId.split(",");
-		assert(parts.length === 3, "wrong channel ID");
+		assert(parts.length === 3, "Invalid channel Id");
 		const rowId = parts[0];
 		const colId = parts[1];
 		const iteration = parts[2];
+		return { rowId, colId, iteration };
+	}
 
-		// TBD(Pri0)
-		// This needs to be way more efficient by building a reverse mapping and maintaining it through all operations!
-		// Need to reconcile which layer does such mapping. At the moment, this is done outside of matrix - in Tablero
-		// component. We should either move it here (and ideally - in compatible way, i.e. not changing document schema),
-		// or move most of the channel logic to the layer that implements row/col stable IDs.
-
-		const rowCount = this.matrix.rowCount;
-		const colCount = this.matrix.colCount;
-
-		let row;
-		for (row = 1; row < rowCount; row++) {
-			if (this.areEqualUuid(this.matrix.getCell(row, 0) as unknown as uuidType, rowId)) {
-				break;
-			}
-		}
-		assert(row !== rowCount, "channel not found");
-
-		let col;
-		for (col = 1; col < colCount; col++) {
-			if (this.areEqualUuid(this.matrix.getCell(0, col) as unknown as uuidType, colId)) {
-				break;
-			}
-		}
-		assert(col !== colCount, "channel not found");
+	private mapChannelToCell(channelId: string): { row: number; col: number; iteration: string } {
+		const { rowId, colId, iteration } = this.parseChannelId(channelId);
+		const row = this.reverseMap.getRowId(rowId);
+		const col = this.reverseMap.getColId(colId);
 
+		assert(
+			row !== undefined &&
+				this.areEqualUuid(this.matrix.getCell(row, 0) as unknown as uuidType, rowId),
+			"channel's row not found",
+		);
```

Comment:
I was wrong here - please see latest proposed changes in https://github.com/vladsud/FluidFramework/pull/13/files
I do not care in what order we merge our PRs (I can fix it), but this function needs to deal properly (not crash or assert) when it can't map channel to cell.